source/parser.hera

```diff
@@ -3628,7 +3640,22 @@ CatchParameter
 
 # An expression with explicit or implied parentheses, for use in if/while/switch
 Condition
+  # NOTE: DeclarationCondition must come before the ExpressionCondition because we need to look ahead to potentially match `:=` or `.=`
+  OpenParen:open _?:ws DeclarationCondition:expression CloseParen:close ->
+    return {
+      type: "ParenthesizedExpression",
+      children: [open, ws, expression, close],
+      expression,
+    }
   ParenthesizedExpression !( _? ( BinaryOp / AssignmentOp / Dot / QuestionMark ) ) !( _ OperatorAssignmentOp ) -> $1
+  # NOTE: DeclarationCondition must come before the ExpressionCondition because we need to look ahead to potentially match `:=` or `.=`
+  InsertOpenParen:open DeclarationCondition:expression InsertCloseParen:close ->
+    return {
+      type: "ParenthesizedExpression",
+      children: [open, expression, close],
+      expression,
+    }
+
```

Comment:
Hmm, I expected this to be allowed in an arbitrary Expression, so you could do things like `if x + (y := ...)` but I guess that can be an extension for later.